job_runner/worker.py

```
@@ -0,0 +1,167 @@
+
+
+async def publish_failed_status(redis_client: redis.Redis, job_id: str, error: str):
+    update_payload = {
+        "status": "failed",
+        "current_step": "Worker failed",
+        "error": error,
+        "messages": [
+            {
+                "role": "agent",
+                "content": "Agent job failed before completion. Check worker logs.",
+                "timestamp": datetime.now().isoformat(),
```

Ruff (0.15.10)

[warning] `datetime.datetime.now()` called without a `tz` argument

(DTZ005)